gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/CircularDependencyChecker.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.Path;
+
+
+public class CircularDependencyChecker {
```

Comment:
The methods in this class just check whether a particular URI is a seed for a cycle. I am assuming the intention of this is to first check for cycles, then do resolution? It should be possible to check for cycles at the same time as resolution is being computed.
I think it would be useful to have a method to check for cycles in the entire store, without requiring a seed. You can use Strongly connected component finding algorithms to do this ( for example https://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm ).